acceptance: adding repair test

BramGruneir opened the pull request:
This commit adds a dynamic client which is designed to handle clusters with
changing shapes. So it is adaptable to adding and removing nodes.

Also, this includes the first isolated load. This is the first of many loads
that we will be able to apply to tests in a more generic way. This load is
designed to be a non-contentious one that just inserts into a table and checks
that the values in the table are still correct.

The repair test so far is only an insert test. I will be following this up with
changes to cluster to handle adding and permanently removing nodes.

This is part of #5345.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/5481)

<!-- Reviewable:end -->

BramGruneir committed:
acceptance: adding repair test

This commit adds a dynamic client which is designed to handle clusters with
changing shapes. So it is adaptable to adding and removing nodes.

Also, this includes the first isolated load. This is the first of many loads
that we will be able to apply to tests in a more generic way. This load is
designed to be a non-contentious one that just inserts into a table and checks
that the values in the table are still correct.

The repair test so far is only an insert test. I will be following this up with
changes to cluster to handle adding and permanently removing nodes.

This is part of #5345.

tamird commented:
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 33 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV3XN4jmNAffnvklBT:-KDV3XN4jmNAffnvklBU:1859865044) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L33)):*
as mentioned elsewhere, this doesn't compose well :(

---

*[acceptance/dynamic_client.go, line 35 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1CzLpbolj6LP4BdR:-KDV1CzMKQxnqTl6Lxyb:1566347855) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L35)):*
this seemsoverly complicated - why do callers need to know about underlying clients at all? perhaps you could hang the Cluster on this guy, and just have a few methods that proxy to an underlying client which is managed entirely by this structure?

---

*[acceptance/dynamic_client.go, line 102 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1z5ISUf8uEJC5yl1:-KDV1z5J2KK1ci4SyTHa:-430715711) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L102)):*
this code already exists somewhere.

---

*[acceptance/load.go, line 27 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV3KcQiHUZatY-qfH2:-KDV3KcRI9mXW9bvBzHL:749262275) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/load.go#L27)):*
i think i'd make this take a `testing.T` rather than passing it through the dynamicClient

---

*[acceptance/repair_test.go, line 53 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV0c7d83gRY2YROtj3:-KDV0c7d83gRY2YROtj4:-1809554108) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/repair_test.go#L53)):*
wow, so this is actually a busy loop. how about the slightly less busy
```
ctx, cancel := context.WithTimeout(context.Background(), cfg.Duration)
defer cancel()
select {
case <-stopper:
case <-ctx.Done():
}
```

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 31 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDVBX9mtDCE1B-yNPWl:-KDVBX9nXGZjKULSpa5f:-963573741) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L31)):*
I think it will be valuable to add comments for each of these fields. Do you want to rename dynamicClient to perhaps clientManager?

---

*[acceptance/dynamic_client.go, line 33 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV3XN4jmNAffnvklBT:-KDV9zSGtP2o5ywcByAz:-62460207) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L33)):*
isRunning()

---

*[acceptance/dynamic_client.go, line 37 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDVC5fQZBFsiSA29TDQ:-KDVC5fQZBFsiSA29TDR:-1032759771) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L37)):*
Why do you need to store the URLs? Is there value to defining the interface used by the users of this manager, and how that might get affected by nodes coming and going.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 1 of 4 files reviewed at latest revision, 7 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 31 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDVBX9mtDCE1B-yNPWl:-KDVDFbVTfr5AdvJYajF:1527014678) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L31)):*
It's a db client manager, or can be seen as a dynamic client to it's consumers.  
I've renamed a good amount of the fields and added comments, it should be clearer now.

---

*[acceptance/dynamic_client.go, line 33 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV3XN4jmNAffnvklBT:-KDZ3BjkV6W8GuWeYQmE:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L33)):*
Done.

---

*[acceptance/dynamic_client.go, line 35 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1CzLpbolj6LP4BdR:-KDVCKT0F2-FTdfP5sZi:-1374275601) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L35)):*
That's exactly what I'm trying to do with this struct.  Instead of using getClient, use exec() and queryRowScan().  No need to deal with the client at all.  I didn't think about adding it to the cluster, that might work, but I think I'd like to do that after beta as a bigger refactor.  My goal right now is minimum disruption while adding in new functionality.

---

*[acceptance/dynamic_client.go, line 37 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDVC5fQZBFsiSA29TDQ:-KDZ3EvysNbTkZPmvvk4:-1128338147) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L37)):*
These urls are used internally and randomly chosen from when opening a new connection to the db.  When a connection fails and cannot reconnect, its url is removed from this list.

---

*[acceptance/dynamic_client.go, line 102 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1z5ISUf8uEJC5yl1:-KDVCnqZhvkan0avB1-f:-792263789) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L102)):*
It does.  In chaos test.  Which I'll be removing and porting to the dynamic client in the future.  I don't think this duplication is a big issue.   But I can add a todo if you think it's required.

---

*[acceptance/load.go, line 27 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV3KcQiHUZatY-qfH2:-KDVCyRsDOeC7Jp9MaJw:-1158561979) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/load.go#L27)):*
Sure, can do.

---

*[acceptance/repair_test.go, line 53 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV0c7d83gRY2YROtj3:-KDVDqTs9kPQt9cIcGOf:84449021) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/repair_test.go#L53)):*
I'm thinking just a blocking select.  You're right, no need for the busy loop,

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 35 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1CzLpbolj6LP4BdR:-KDZBFkI0HzKfLmC5wBZ:-1559297914) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L35)):*
I think you misunderstood me - I'm not suggesting you hang this on the Cluster; I'm suggesting you hang a Cluster on this.

---

*[acceptance/dynamic_client.go, line 102 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1z5ISUf8uEJC5yl1:-KDZBPprd0aZhBs_bH04:1548294109) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L102)):*
can you just appease my OCD and make that code call this? a TODO for a 3 line change seems heavy.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 1 of 5 files reviewed at latest revision, 5 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 35 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1CzLpbolj6LP4BdR:-KDZR8zqrc3TRuF_HABk:1787957738) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L35)):*
I like this idea a lot.  Going to do this.

---

*[acceptance/dynamic_client.go, line 102 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1z5ISUf8uEJC5yl1:-KDZGwHvjhS2QO6GU0Wa:-1719105008) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L102)):*
sure, done

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
I'm going to do as suggested and have the dynamic client query the cluster directly.  I'll update once this is done.

---

Review status: 1 of 5 files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-:-KDZROTckZqwmyeWKToI:638682756)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 3 of 3 files at r3, 3 of 3 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 102 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1z5ISUf8uEJC5yl1:-KDZSb5Kz8LyLznG3Ccj:-1676283691) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L102)):*
The version of this code that you removed is terser - perhaps use it rather than call `testutils.IsError` n times?
```
return testutils.IsError(err, "connection reset by peer") || testutils.IsError(err, "connection refused") || testutils.IsError(err, "failed to send RPC") || testutils.IsError(err, "EOF")
```

by the way, some of these errors are indeed retryable (perhaps only "failed to send RPC") rather than reconnectable. perhaps keep that in mind as you rework this - that'll allow you to avoid spurious reconnections.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[acceptance/dynamic_client.go, line 102 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1z5ISUf8uEJC5yl1:-KDZVdizSMwp2loEPXLJ:-1192106222) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L102)):*
In what way is it terser?  It's the same code, mine is just spaced a little differently.
Also, yes, I am fully aware the this list is going to change.  I thought it might diverge, which is why I was thinking of having two copies.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

tamird commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[acceptance/dynamic_client.go, line 102 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1z5ISUf8uEJC5yl1:-KDZWO8Fe2MWyAE4cWOW:1636563990) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L102)):*
oh, apologies, I was thinking of the version of this code which was apparently never merged https://github.com/cockroachdb/cockroach/pull/5440/files

that version constructs just one regular expression rather than 4.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Ok, all set, PTAL.

---

Review status: 1 of 5 files reviewed at latest revision, 4 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 102 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDV1z5ISUf8uEJC5yl1:-KDZtNK8gCDaJUD-AghV:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/def4e1f32a8c857fb19fd8f395526fc9767718b6/acceptance/dynamic_client.go#L102)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-:-KDZtOo1D69S2iXuhnzP:259437405)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 4 of 4 files at r6.
Review status: all files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[acceptance/dynamic_client.go, line 37 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZwyMAHqRP_-Y9rB0F:-KDZwyMAHqRP_-Y9rB0G:-606115545) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L37)):*
why not use an anonymous struct?

---

*[acceptance/dynamic_client.go, line 59 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZx5hriLfMKzRRjK1e:-KDZx5hriLfMKzRRjK1f:1654201146) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L59)):*
is this actually possible?

---

*[acceptance/dynamic_client.go, line 60 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZyGDBOJ6kS66Rs75X:-KDZyGDCfsQq2NebKfED:4783852) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L60)):*
throughout: getHost is really reversing the logic in `PGUrl`. If you really want to log this, export a new method on Cluster (hostname? address? you decide) rather than doing this shenanigan.

Personally, I think we can do without this log line, or we can just log PGUrl.

---

*[acceptance/dynamic_client.go, line 71 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZy08vX5giMEcDgdqA:-KDZy08vX5giMEcDgdqB:1092333727) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L71)):*
"connection reset by peer", "connection refused", "EOF" all of these may indicate that the node has gone away.

---

*[acceptance/dynamic_client.go, line 76 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZyYRdSBW3wjEHxMRU:-KDZyYRdSBW3wjEHxMRV:1555297986) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L76)):*
isRunning is never called with any other argument than `dynamicClient-which-is-in-scope.finished`. make it a method

---

*[acceptance/dynamic_client.go, line 121 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZyomGpc96jmyti-rg:-KDZyomGpc96jmyti-rh:849458507) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L121)):*
why would client ever be nil if `ok` is true?

---

*[acceptance/dynamic_client.go, line 136 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZz0ZJOcvQhMXgcvcY:-KDZz0ZJOcvQhMXgcvcZ:-384839608) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L136)):*
does this do anything?

---

*[acceptance/dynamic_client.go, line 148 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZz8x56ZIF2ZseQaiK:-KDZz8x56ZIF2ZseQaiL:-1155065739) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L148)):*
this should return an error

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: all files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[acceptance/dynamic_client.go, line 37 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZwyMAHqRP_-Y9rB0F:-KD_7zhKIpJAuR-WV6jX:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L37)):*
Done.

---

*[acceptance/dynamic_client.go, line 59 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZx5hriLfMKzRRjK1e:-KD_0Lpl4ExWpg1Zq4K6:1999821805) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L59)):*
It's a map to a pointer, so technically yes. It's just a safeguard.
But yes, it is
https://play.golang.org/p/QqutvkZLtT

---

*[acceptance/dynamic_client.go, line 60 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZyGDBOJ6kS66Rs75X:-KD_1glsuuUFlMn1h108:288572354) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L60)):*
PGUrl is very ugly and contains all the certs and security info.  Exported a new method on cluster, Addr.
The log line is quite useful and they don't actually occur very often.  We can always move it to a v1 log if needed.

---

*[acceptance/dynamic_client.go, line 71 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZy08vX5giMEcDgdqA:-KD_8FbOOL5aaCCfmF0P:1500815095) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L71)):*
Yeah.  The point in the dynamic client is not to retry the query against the same node, but to use another node instead.  I've rewritten the comment to be accurate.  For the chaos test, where this code was originally from, it keeps retrying against the same node (to establish that the node is again healthy).  

---

*[acceptance/dynamic_client.go, line 76 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZyYRdSBW3wjEHxMRU:-KD_3RHF1vpOuK6Ge22m:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L76)):*
Done.

---

*[acceptance/dynamic_client.go, line 121 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZyomGpc96jmyti-rg:-KD_3Ut_AfeBcDB5t-j0:1907991182) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L121)):*
A map to a pointer can store a nil value and will return true.  It shouldn't in this case, but it's an easy enough case to check for.  But, it will default to nil, so I can ignore the ok.

---

*[acceptance/dynamic_client.go, line 136 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZz0ZJOcvQhMXgcvcY:-KD_5j63LVkU2k8AgdMc:766306466) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L136)):*
Yes.  It actually establishes that the connection is active.  See the documentation:
https://golang.org/pkg/database/sql/#Open

---

*[acceptance/dynamic_client.go, line 148 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZz8x56ZIF2ZseQaiK:-KD_8ZkXUAWDtqCH9Bds:2048026321) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L148)):*
Why?  The only time this returns a nil.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 6 of 6 files at r7.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 59 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZx5hriLfMKzRRjK1e:-KD_B6DO_HtgGULNZmFK:1617438185) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L59)):*
heh. Yes, I understand it's a pointer, but if this actually happens that this thing is nil, I would rather see an NPE than this silent swallowing of the mistake.

---

*[acceptance/dynamic_client.go, line 121 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZyomGpc96jmyti-rg:-KD_BWqQG0LIH0hX1BLH:-1973522978) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L121)):*
No, please don't do this. I'd rather we see an NPE here than pokemon this error.

---

*[acceptance/dynamic_client.go, line 136 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZz0ZJOcvQhMXgcvcY:-KD_BnPW6h1xFy0qMQCG:-1138511539) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L136)):*
Huh, it doesn't even require anything from the driver author. TIL.

---

*[acceptance/dynamic_client.go, line 148 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZz8x56ZIF2ZseQaiK:-KD_BqvoKv1PSrs6Lctm:-522765599) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L148)):*
Because that's not how error handling is done in Go. This method returns a pointer because that's what db.Open (sadly) returns, not because nullability is desired.

---

*[acceptance/dynamic_client.go, line 32 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_B3WqbMif99Z6r55n:-KD_B3WqbMif99Z6r55o:1562688405) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L32)):*
I thought this was going away?

---

*[acceptance/dynamic_client.go, line 37 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_AobX2Jl7rW-wpaks:-KD_Aobvs4qqJSQVSH69:-2129293997) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L37)):*
comment is not useful

---

*[acceptance/dynamic_client.go, line 38 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_AsAcvwU3fOjNFZS5:-KD_AsAcvwU3fOjNFZS6:1470542971) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L38)):*
be explicit about these indexes matching the indexes passed to methods on the Cluster

---

*[acceptance/dynamic_client.go, line 68 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_BOft_IHnY7TquzQz:-KD_BOfug1dL0TQJgQb5:-571068966) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L68)):*
s/possibly //

---

*[acceptance/dynamic_client.go, line 69 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_BQenfjCxrJALSMrz:-KD_BQenfjCxrJALSMs-:1061547401) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L69)):*
s/resetting/restarting/

---

*[acceptance/dynamic_client.go, line 128 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_CHqFNOaVXEcHAmyR:-KD_CHqFNOaVXEcHAmyS:1363337626) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L128)):*
remove this for reasons discussed elsewhere

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 59 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZx5hriLfMKzRRjK1e:-KD_GgpYBIGVy3r-lyip:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L59)):*
Done.

---

*[acceptance/dynamic_client.go, line 121 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZyomGpc96jmyti-rg:-KD_EtDgZ_kJ0TqpxYXE:-1130189104) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L121)):*
done.  But what does pokemon mean as a verb?

---

*[acceptance/dynamic_client.go, line 148 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZz8x56ZIF2ZseQaiK:-KD_Fx-mk3Ri7bIt0i3s:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L148)):*
Done.

---

*[acceptance/dynamic_client.go, line 32 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_B3WqbMif99Z6r55n:-KD_GcrniTf7mJXzwJ5o:849868843) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L32)):*
so did I, removed.

---

*[acceptance/dynamic_client.go, line 37 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_AobX2Jl7rW-wpaks:-KD_E--Hy2bpryGKRcXD:-1929989507) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L37)):*
Pretty standard comment from other anonymous structs in our codebase, but you're right, removed.

---

*[acceptance/dynamic_client.go, line 38 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_AsAcvwU3fOjNFZS5:-KD_EY2xnoEvQDNz2NwT:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L38)):*
Done.

---

*[acceptance/dynamic_client.go, line 68 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_BOft_IHnY7TquzQz:-KD_EhJOTqpCY8Rzls18:-221146732) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L68)):*
ick, fixed

---

*[acceptance/dynamic_client.go, line 69 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_BQenfjCxrJALSMrz:-KD_EidtXDR0Cg2JGDZK:-1430021183) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L69)):*
double ick.  fixed too

---

*[acceptance/dynamic_client.go, line 128 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_CHqFNOaVXEcHAmyR:-KD_F5orDWHOUAOSXfSB:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/d77ad2df4d6af74c7572eb61311e83ee35af59af/acceptance/dynamic_client.go#L128)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 2 of 2 files at r8.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 59 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZx5hriLfMKzRRjK1e:-KD_I8v9MsLfqodpvmOl:163023921) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L59)):*
looks like you're still nil checking?

---

*[acceptance/dynamic_client.go, line 121 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZyomGpc96jmyti-rg:-KD_JHCypsDduhzmScEh:2058738799) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L121)):*
gotta catch 'em all.

---

*[acceptance/dynamic_client.go, line 81 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_IS-gjKHdNJJwWnCS:-KD_IS-gjKHdNJJwWnCT:-290184031) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L81)):*
```
if err == nil || !isRetryableError(err) {
  return result, err
}
```

---

*[acceptance/dynamic_client.go, line 100 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_IinHGdUZysquL8rl:-KD_IinHGdUZysquL8rm:-469765787) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L100)):*
ditto

---

*[acceptance/dynamic_client.go, line 118 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_Iu6PzkAoq-BCrk3m:-KD_Iu6PzkAoq-BCrk3n:418309215) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L118)):*
why not do the less-indented
```
if client, ok := dc.mu.clients[index]; ok {
return client, nil
}
```

looks like maybe you want an inner function for this so you can keep the ping logic you're doing

---

*[acceptance/dynamic_client.go, line 132 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_JaCPJ2kJitLKI9fh:-KD_JaCPJ2kJitLKI9fi:-1936641297) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L132)):*
there's something dubious about this since clients are internally multiplexed. perhaps worth investigating a bit further.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 77 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDaa585bD4Wpod-VDI3:-KDaa586fDv1BD0WlSJx:1246745562) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L77)):*
You cannot receive a retryable error here and in queryRowScan

---

*[acceptance/dynamic_client.go, line 116 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDaaHoVqo12QBXva03l:-KDaaHoVqo12QBXva03m:-1786442885) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L116)):*
randomized load balancing, excellent

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 59 \[r6\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDZx5hriLfMKzRRjK1e:-KD_K6NvNPwvu4E4nfc0:-2002186940) ([raw file](https://github.com/cockroachdb/cockroach/blob/748f2dc2a216252e43ba1f1333dad1d8b0bfa09e/acceptance/dynamic_client.go#L59)):*
bah, I did remove it, not sure what happened.

---

*[acceptance/dynamic_client.go, line 77 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDaa585bD4Wpod-VDI3:-KDd0XfvfoLf9FHbCYbk:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L77)):*
Done.

---

*[acceptance/dynamic_client.go, line 81 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_IS-gjKHdNJJwWnCS:-KD_KNXmC04hSCpR6wKy:677880290) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L81)):*
nice

---

*[acceptance/dynamic_client.go, line 100 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_IinHGdUZysquL8rl:-KD_KTTFD_Psh3HTrnwI:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L100)):*
Done.

---

*[acceptance/dynamic_client.go, line 118 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_Iu6PzkAoq-BCrk3m:-KD_KhCb5j0bBgfvGwDo:2052721482) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L118)):*
Removed the ping logic.

---

*[acceptance/dynamic_client.go, line 132 \[r8\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KD_JaCPJ2kJitLKI9fh:-KD_OZTa_MaibmaVD_v4:2087930920) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4251edf0d4423100b9de7dd5a24af7d6aec30c6/acceptance/dynamic_client.go#L132)):*
Removed the Ping.  

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 2 of 2 files at r9.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 32 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd3tcEqUvNOLDDU7gK:-KDd3tcEqUvNOLDDU7gL:1675705178) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L32)):*
nit: letters are cheap. call this `cluster` and call the stopper `stopper`

---

*[acceptance/dynamic_client.go, line 80 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd3hrVQjlLUhnrcizp:-KDd3hrWwyHM3FPSxyXV:708786261) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L80)):*
optional nit: you could inline this into the line below

---

*[acceptance/dynamic_client.go, line 85 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd1y-gZgtQ-LjO4-sL:-KDd1y-hpof1CKcfLVBc:-722360647) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L85)):*
You need to return an error here. Returning `nil,nil` is extremely likely to blow up the caller.

---

*[acceptance/dynamic_client.go, line 96 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd3keqXqkRPB0xn1Bv:-KDd3keqXqkRPB0xn1Bw:708786261) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L96)):*
optional nit: you could inline this into the line below

---

*[acceptance/dynamic_client.go, line 101 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd2nkkmNSpJM9DFQLf:-KDd2nkkmNSpJM9DFQLg:-1860333856) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L101)):*
ditto, you need to return an error here

---

*[acceptance/dynamic_client.go, line 129 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd3LYs5dwC6aZiejBF:-KDd3LYs5dwC6aZiejBG:-82726989) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L129)):*
Remove this function. Every caller of it can be replaced with

```
for dc.s.RunTask(func() {
....
})
```

---

*[acceptance/repair_test.go, line 33 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd2D4Ag4MaTlIu9MdV:-KDd2D4Ag4MaTlIu9MdW:-811971557) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/repair_test.go#L33)):*
nit: don't shadow package names - call this `stopper` like we do everywhere else, and `defer` the `Stop`.

---

*[acceptance/repair_test.go, line 48 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd2QSGaGuP-gRe0pDF:-KDd2QSGaGuP-gRe0pDG:753802164) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/repair_test.go#L48)):*
undefined?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 5 of 8 files reviewed at latest revision, 10 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 32 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd3tcEqUvNOLDDU7gK:-KDd5z3gUpzxsGg3gW9Z:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L32)):*
Done.

---

*[acceptance/dynamic_client.go, line 80 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd3hrVQjlLUhnrcizp:-KDd69SpsF47vovH7sI-:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L80)):*
Done.

---

*[acceptance/dynamic_client.go, line 85 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd1y-gZgtQ-LjO4-sL:-KDd7YbNGwMJzwoeVm2N:-72118368) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L85)):*
done, added a new error and check for it in insertLoad to not panic the test during shutdown

---

*[acceptance/dynamic_client.go, line 96 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd3keqXqkRPB0xn1Bv:-KDd8jZrC-LplowU6WrQ:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L96)):*
Done.

---

*[acceptance/dynamic_client.go, line 101 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd2nkkmNSpJM9DFQLf:-KDd8kUYhSKudVtsOXZ4:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L101)):*
Done.

---

*[acceptance/dynamic_client.go, line 129 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd3LYs5dwC6aZiejBF:-KDdhjN7t9hAzVn3JsMk:832275259) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L129)):*
Sadly, it just makes it look worse.  I'm going to leave this as is.

---

*[acceptance/repair_test.go, line 33 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd2D4Ag4MaTlIu9MdV:-KDd4tuWZtBoyzW6f22J:-1135423127) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/repair_test.go#L33)):*
yeah, I didn't mean to shadow the name.  Sadly, we have another stopper that's used for interruptions.  I'll give it a new name.

---

*[acceptance/repair_test.go, line 48 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd2QSGaGuP-gRe0pDF:-KDd5AIqLq-R9kDUMGx4:-2115891886) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/repair_test.go#L48)):*
This stopper is a global in the acceptance tests (just a `chan struct{}`) used for detecting interruptions.  I plan to refactor it to use a real stopper, but haven't had a chance yet.  If this didn't exist, I would have used the name stopper like everywhere else.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r10.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[acceptance/dynamic_client.go, line 96 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDd3keqXqkRPB0xn1Bv:-KDeG-w_0HZfFmkJ30h-:-1079716148) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7662b4629f727d3fca8426f4133f90ff15b3898/acceptance/dynamic_client.go#L96)):*
prefer `:=` to `=` (that was the point of inlining in my mind)

---

*[acceptance/load.go, line 70 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDeGIee-ICxSGgPdHgk:-KDeGIee-ICxSGgPdHgl:-1530322978) ([raw file](https://github.com/cockroachdb/cockroach/blob/2b0c40c47550655794bff032926d8bf90817b99f/acceptance/load.go#L70)):*
inline

---

*[acceptance/repair_test.go, line 51 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-KDeGRUoJGnwlKiyHuJH:-KDeGRUoJGnwlKiyHuJI:-1149131158) ([raw file](https://github.com/cockroachdb/cockroach/blob/2b0c40c47550655794bff032926d8bf90817b99f/acceptance/repair_test.go#L51)):*
why not defer like everything else?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5481#-:-KDeGxQlzKuTm0qbgWy7:1417088721)*
<!-- Sent from Reviewable.io -->
